packages/workflow/test/Workflow.test.ts

```diff
@@ -68,6 +69,84 @@ describe('Workflow', () => {
 		});
 	});
 
+	describe('checkReadyForExecution', () => {
+		const disabledNode = mock<INode>({ name: 'Disabled Node', disabled: true });
+		const startNode = mock<INode>({ name: 'Start Node' });
+		const unknownNode = mock<INode>({ name: 'Unknown Node', type: 'unknownNode' });
+
+		const nodeParamIssuesSpy = jest.spyOn(NodeHelpers, 'getNodeParametersIssues');
+
+		const nodeTypes = mock<INodeTypes>();
+		nodeTypes.getByNameAndVersion.mockImplementation((type) => {
+			// TODO: getByNameAndVersion signature needs to be updated to allow returning undefined
+			if (type === 'unknownNode') return undefined as unknown as INodeType;
+			return mock<INodeType>({
+				description: {
+					properties: [],
+				},
+			});
+		});
+
+		beforeEach(() => jest.clearAllMocks());
+
+		it('should return null if there are no nodes', () => {
+			const workflow = new Workflow({
+				nodes: [],
+				connections: {},
+				active: false,
+				nodeTypes,
+			});
+
+			const issues = workflow.checkReadyForExecution();
+			expect(issues).toBe(null);
+			expect(nodeTypes.getByNameAndVersion).not.toHaveBeenCalled();
+			expect(nodeParamIssuesSpy).not.toHaveBeenCalled();
+		});
+
+		it('should return null if there are no enabled nodes', () => {
+			const workflow = new Workflow({
+				nodes: [disabledNode],
+				connections: {},
+				active: false,
+				nodeTypes,
+			});
+
+			const issues = workflow.checkReadyForExecution({ startNode: disabledNode.name });
```

Comment:
You're given a `startNode` in multiple tests here, but no `destinationNode`.
All these tests would also work if we didn't pass the start node. So I assume it's to be able to test the branch that calls `getChildNodes`.
Would it make sense to change one of these tests to pass `destinationNode` so that we also cover `getParentNodes`?

Reply:
I think we should add a separate group of tests for `destinationNode`. 
But, I can do that in the next PR adding tests for `Workflow`.